### DISCUSSION TITLE
Write only atom that returns (prevState) => {} - setAction

### DISCUSSION THREAD

**Discussion by AlekseyP18** [pmndrs/jotai#2553] [Q&A] [answered]
Hello, I am new in using jotai user)
How can I create a writeOnlyAtom that will have setAction like basic atom? Thanks)

const write = ....;
const userWriteAtom = atom(null, write)

const [, setUser] = useAtom(userWriteAtom);

const updateUser = (newUser) => setUser(prevUser => ({ ...prevUser, ...newUser }));
const resetUser = () => setUser(null);

**Comment by dai-shi** [comment, marked as answer]
There are some ways depending on your requirement with the write-only atom, but this should help:
```js
const userWriteAtom = atom(null, (get, set, arg) => {
  const prevUser = get(currentUserAtom);
  const nextUser = typeof arg === 'function' ? arg(prevUser) : arg;
  set(currentUserAtom, nextUser);
});
```

**Reply by AlekseyP18** [reply]
Thanks for fast answer! Here is a good community)
